gapic-generator/lib/gapic/model/method/http_annotation.rb

```diff
@@ -35,122 +43,181 @@ class HttpAnnotation
+            verb_path[0] if verb_path
+          end
+
+          ##
+          # Whether a method path is present and non-empty
+          #
+          # @return [Boolean]
+          def path?
+            !path.empty?
+          end
+
+          ##
+          # A method path or an empty string if not present
+          #
+          # @return [String]
+          def path
+            return "" unless verb_path
+            verb_path[1]
+          end
+
+          ##
+          # Whether any routing params are present
+          #
+          # @return [Boolean]
+          def routing_params?
+            routing_params.any?
+          end
+
+          ##
+          # The segment key names and their corresponding paterns,
+          # including the `*` pattern implied for the named segments
+          # without pattern explicitly specified.
+          #
+          # @return [Array<Array<String>>]
+          def routing_params_with_patterns
+            @routing_params_with_patterns ||= begin
+              Gapic::UriTemplate.parse_arguments(path).map do |name, pattern|
+                [name, pattern.empty? ? "*" : pattern]
+              end
             end
           end
-        end
 
-        ##
-        # The segment key names.
-        #
-        # @return [Array<String>]
-        def routing_params
-          routing_params_with_patterns.map { |param, _| param }
-        end
+          ##
+          # The segment key names.
+          #
+          # @return [Array<String>]
+          def routing_params
+            routing_params_with_patterns.map { |param, _| param }
+          end
 
-        ##
-        # Whether method has body specified in proto
-        #
-        # @return [Boolean]
-        def body?
-          !body.empty?
-        end
+          ##
+          # Whether method has body specified in proto
+          #
+          # @return [Boolean]
+          def body?
+            !body.empty?
+          end
 
-        ##
-        # The body specified for the given method in proto
-        # or an empty string if not specified
-        #
-        # @return [String]
-        def body
-          http&.body || ""
+          ##
+          # The body specified for the given method in proto
+          # or an empty string if not specified
+          #
+          # @return [String]
+          def body
+            @binding&.body || ""
+          end
+
+          private
+
+          ##
+          # The combination of verb and path found in the http annotation
+          # (or Nil if the annotation is Nil).
+          #
+          # @return [Array<Symbol, String>, Nil]
+          def verb_path
+            return nil if @binding.nil?
+
+            method = {
+              get:    @binding.get,
+              post:   @binding.post,
+              put:    @binding.put,
+              patch:  @binding.patch,
+              delete: @binding.delete
+            }.find { |_, value| !value.empty? }
+
+            method unless method.nil?
```

Comment:
`unless method.nil?` is redundant. If `method.nil?` then the statement will fall through and return `nil` anyway.

Which in turn implies we can dispense with the local `method` variable altogether and just directly return the result of the `find` call above.

Reply:
:+1: